### ISSUE TITLE
Generation of API sequence tests

### ISSUE THREAD

**Issue by noahdietz** [apigee-127/swagger-test-templates#98]
I want to add an option that provides the ability to create tests containing a sequence of API calls. A common design paradigm in REST APIs is CRUD (Create, Retrieve, Update and Delete). As such, it is important to test these sequences (or ones specified by the developer) to ensure the core use case is operational.

This is not limited to CRUD, but will be configurable by providing a sequence object at test generation.
Below is an example of the proposed option, for the CRUD sequence, identified by the name 'seq1':
```js
testGen(swagger, {
      testModule: 'supertest',
      assertionFormat: 'should',
      pathName: [],
      sequences: {
        'seq1': [
          { path: '/user',
            op: 'POST' },
          { path: '/user/{id}',
            op: 'GET' },
          { path: '/user/{id}',
            op: 'PUT' },
          { path: '/user/{id}',
            op: 'DELETE' }
        ]
    }
});
```

**Comment by 0x1mason** [comment]
@noahdietz Does this include the ability to "build" on data obtained in a previous test? E.g., I want to 
1. Get `/foo/users?name=0x1mason` <== returns userId
2. Get `/foo/{userId}/info` <== cherry pick firstName from response body
3. etc

**Comment by noahdietz** [comment]
@0x1mason yes! If you look at the PR right above your comment, I implemented integration test generation in my own fork. The implementation is slightly different than defined in my original issue here. This feature is slightly limited in the support of the robust OpenAPI (fka Swagger), I wanted it as a PoC.

The [README](https://github.com/noahdietz/swagger-test-templates) of my forked repo has the updated usage. 

In reference to your mention of "cherry picking" the dependencies, I implemented it as a shared (global) object. So each request in the sequence will store their response body in the shared object (whether they are a dependency or not), keyed by their _step_ label. These will be accessible to all following requests in the defined sequence.
